### DISCUSSION TITLE
How to deploy an Azure Function using Terraform and AZD

### DISCUSSION THREAD

**Discussion by sebastian-kucharzyk-tr** [Azure/azure-dev#4442] [Q&A]
TL;DR
Did anyone have luck deploying an API to Azure Functions using AZD and can share their code?

Hi, I'm trying to deploy an Azure function based on Python using Terraform and AZD, however, I can't make that work. I'm trying to base my solution on [fastapi-on-azure-functions](https://github.com/Azure-Samples/fastapi-on-azure-functions/tree/main). The key parts I have in terraform are:
1. The function app itself:
```tf
resource "azurerm_linux_function_app" "api" {
  name                = azurecaf_name.fa_name.result
  resource_group_name = var.rg_name
  location            = var.location

  storage_account_name       = azurerm_storage_account.sa.name
  storage_account_access_key = azurerm_storage_account.sa.primary_access_key
  service_plan_id            = azurerm_service_plan.sp.id
  functions_extension_version = "~4"
  https_only = true
  app_settings = {
    FUNCTIONS_WORKER_RUNTIME       = "python"
    SCM_DO_BUILD_DURING_DEPLOYMENT = "true"
    AzureWebJobsFeatureFlags       = "EnableWorkerIndexing"
    ENABLE_ORYX_BUILD = true
  }
  site_config {
    application_stack {
      python_version = "3.11"
    }
  }
  tags=merge(var.tags, { azd-service-name : "api" })
}
```
main.tf:
```tf
module "api" {
  source           = "./modules/api"
  location         = var.location
  rg_name          = module.resource_group.RESOURCE_GROUP_NAME
  environment_name = var.environment_name
  app_name         = var.app_name
  tags             = local.tags
}
```
And the azure.yaml:
```
# yaml-language-server: $schema=https://raw.githubusercontent.com/Azure/azure-dev/main/schemas/alpha/azure.yaml.json

name: my-app
workflows:
  up:
    steps:
      - azd: provision
      - azd: deploy --all
services:
  api:
    project: .
    language: py
    host: function
infra:
  provider: terraform
```
Unfortunately, after deployment, the Function app is created, however, the function itself is not running. Even in Azure Platform I can see just the placeholder to set it up. When I execute the bicep template shared above, everything works fine 🤔.
![image](https://github.com/user-attachments/assets/1f08a0e8-3161-40e4-ab38-9953101d2bd9)
